python/ray/util/client/server/server.py

```diff
@@ -63,36 +62,24 @@ def Init(self, request, context=None) -> ray_client_pb2.InitResponse:
             job_config = pickle.loads(request.job_config)
             job_config.client_job = True
         else:
-            job_config = None
+            job_config = JobConfig()
         current_job_config = None
         with disable_client_hook():
             if ray.is_initialized():
                 worker = ray.worker.global_worker
                 current_job_config = worker.core_worker.get_job_config()
             else:
                 self.ray_connect_handler(job_config)
-        if job_config is None:
-            return ray_client_pb2.InitResponse()
-        job_config = job_config.get_proto_job_config()
-        # If the server has been initialized, we need to compare whether the
-        # runtime env is compatible.
-        if current_job_config and set(job_config.runtime_env.uris) != set(
-                current_job_config.runtime_env.uris):
+
+        # JobConfig sets parameters for the tasks and actors spawned by this
+        # driver, so we must have an exact match here.
```

Comment:
@iycheng I think we were actually not being strict enough here -- it needs to match exactly

Reply:
We could possibly address this in the future by having the client explicitly override all of the options and always have an empty jobconfig

Reply:
Yes, I only check runtime env and make everything else goes with the old path to make sure there is no regression here. I don't like None and I prefer empty JobConfig if we always create an empty one.